proto/v1/structure/languages.proto

```diff
@@ -0,0 +1,86 @@
+syntax = "proto3";
+
+package org.oppia.proto.v1.structure;
+
+import "proto/v1/structure/versions.proto";
+
+option java_package = "org.oppia.v1.structure";
+option java_multiple_files = true;
+
+// Structure corresponds to LanguageStructuresVersion.
+enum LanguageCode {
+  LANGUAGE_CODE_UNSPECIFIED = 0;
+  ENGLISH = 1;
+  ARABIC = 2;
+  HINDI = 3;
+  HINGLISH = 4;
+}
+
+message SubtitledHtml {
+  LanguageStructuresVersion structure_version = 1;
+
+  string content_id = 2;
+  string html = 3;
+}
+
+message RecordedVoiceovers {
+  LanguageStructuresVersion structure_version = 1;
+
+  repeated VoiceoverContentMapping voiceover_language_mapping = 2;
+}
+
+// Structure corresponds to LanguageStructuresVersion.
+message VoiceoverContentMapping {
+  LanguageCode language_code = 1;
+  map<string, Voiceover> voiceover_content_mapping = 2;
+}
+
+// Structure corresponds to LanguageStructuresVersion.
+message Voiceover {
+  string filename = 1;
+  int32 file_size_bytes = 2;
+  reserved 3; // needs_update
```

Comment:
When would a client care about needs_update?

I actually meant to remove all reserved lines, I had just missed this file earlier.

Reply:
needs_update is for when the voiceover is a bit out of sync with the text, because the text has changed and the voiceover isn't updated yet.

(Up to you if you want to keep it in for this iteration or not. I don't have a strong preference.)

Reply:
We don't have a clear UX flow for when that condition is hit. If anything, we'd consider not pushing updates that point to out-of-date voiceovers/text translations to the client in the first place if we're concerned with consistency (which is something we need to figure out as part of the versioning project). The fewer decisions the client has to make, the better I think.

Reply:
I had updated it. Removed all the reserved fields as per the Open task list.